mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -158,40 +158,57 @@ object MesosClient extends StrictLogging with StrictLoggingFlow {
   private val eventDeserializer: Flow[ByteString, Event, NotUsed] =
     Flow[ByteString].map(bytes => Event.parseFrom(bytes.toArray))
 
-  private def connectionSource(frameworkInfo: FrameworkInfo, url: URL)(implicit as: ActorSystem) = {
+  private def connectionSource(frameworkInfo: FrameworkInfo, url: URL, authorization: Option[CredentialsProvider])(
+      implicit as: ActorSystem) = {
     val body = newSubscribeCall(frameworkInfo).toByteArray
 
-    val request = HttpRequest(
-      HttpMethods.POST,
-      uri = Uri("/api/v1/scheduler"),
-      entity = HttpEntity(ProtobufMediaType, body),
-      headers = List(headers.Accept(ProtobufMediaType)))
-
-    val httpConnection = Http().outgoingConnection(url.getHost, url.getPort)
+    def createPostRequest(bytes: Array[Byte], maybeCredentials: Option[HttpCredentials]): HttpRequest =
+      HttpRequest(
+        HttpMethods.POST,
+        uri = Uri(s"${url.getPath}/api/v1/scheduler"),
+        entity = HttpEntity(ProtobufMediaType, bytes),
+        headers = headers.Accept(ProtobufMediaType) :: maybeCredentials.map(Authorization(_)).toList
+      )
+
+    val port = if (url.getPort == -1) url.getDefaultPort else url.getPort
+    val isSecured: Boolean = url.getProtocol == "https"
```

Comment:
wdyt:
```suggestion
    val httpConnection = url.getProtocol match {
      case "https" => Http().outgoingConnectionHttps(url.getHost, port)
      case _ => Http().outgoingConnection(url.getHost, port)
    }
```
```

Reply:
We actually have another implementation where isSecured is used elsewhere.

I would prefer a shared method for detecting this, one that throws an exception or returns an empty optional if that string is neither "https" nor "http"